File: mau/20274.py
def ucln(a, b):
    while b != 0:
        r = a % b
        a = b
        b = r
    return a
def bcnn(a, b):
    return a // ucln(a, b) * b
a, b, c = [int(x) for x in input().split()]
d = bcnn(a,b)
e = bcnn(c,d)
print(e)

File: mau/20274.cpp
#include <iostream>
using namespace std;
int ucln(int m, int n) {
    int r;
    while (n!=0) {
        r = m % n;
        m = n;
        n = r;
    }
    return m;
}
int bcnn(int a, int b) {
    return a / ucln(a,b) * b;
}
int main()
{
    int a, b, c;
    cin >> a >> b >> c;
    int d = bcnn(a, b);
    int e = bcnn(c, d);
    cout << e;
    return 0;
}
